### PR TITLE
Load resource status checks from client-side

### PR DESCRIPTION
**Pull request by davet1985**
* each status bar is a VueJS component
* a set number of retries with a delay between
* retry button if the checks can still not be fetched after the initial tries
* modifying the VueJS component registration to handle multiple instances of the same component on a page
* moving status checks code from `resource_helper` to a new service and exposing as a JSON endpoint

### CODE REVIEW

File: app/agents/git_hub_agent.rb [hunk 1]
```diff
@@ -161,7 +161,7 @@ def remove_user_from_team(team_name, username)
   def setup_client
     payload = {
       iat: Time.now.to_i,
-      exp: Time.now.to_i + (10 * 60), # Max is 10 mins
+      exp: Time.now.to_i + (9 * 60), # Max is 10 mins
```
Comment (jits) [inline app/agents/git_hub_agent.rb]: Is this change needed? (This is the expiry for the JWT token)

**Comment by jits** [comment]
For completeness, could you update:
- https://github.com/appvia/appvia-hub/blob/master/spec/routing/resources_routing_spec.rb
- https://github.com/appvia/appvia-hub/blob/master/spec/requests/resources_spec.rb

File: app/webpack/packs/application.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ import Vue from 'vue/dist/vue.esm';
 import TurbolinksAdapter from 'vue-turbolinks';
 import Autorefresh from '../components/Autorefresh.vue';
 import AddUserToTeam from '../components/AddUserToTeam.vue';
+import StatusBar from '../components/StatusBar.vue';
```
Comment (jits) [inline app/webpack/packs/application.js]: Just to be super clear, may be good to rename this to `ResourceStatusBar`?

(Aside: it's annoying that we now have another notion of resource status, on top of the already `resource.status` field – I know that wasn't introduced in this PR, but it's becoming more apparent as an issue now, and with me working on https://github.com/appvia/appvia-hub/issues/194).

File: app/agents/git_hub_agent.rb (hunk 1 above)
Comment (davet1985) [reply app/agents/git_hub_agent.rb]: I was about to comment showing the error I was getting, but it's now working fine with this reverted 😕 Will change this back.

File: app/webpack/packs/application.js (hunk 2 above)
Comment (davet1985) [reply app/webpack/packs/application.js]: As discussed offline, this will now be called `ResourceChecks` and all other names using `status` amended also.

**Comment by davet1985** [comment]
> For completeness, could you update:
> 
> * https://github.com/appvia/appvia-hub/blob/master/spec/routing/resources_routing_spec.rb
> * https://github.com/appvia/appvia-hub/blob/master/spec/requests/resources_spec.rb

Updated.
